src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -629,42 +615,82 @@ public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item project) {
             return new StandardListBoxModel();
         }
 
-        public String getStashRootUrl() {
-            if ((stashRootUrl == null) || (stashRootUrl.trim().isEmpty())) {
-                return null;
-            } else {
-                return stashRootUrl;
-            }
-        }
-
-        public boolean isDisableInprogressNotification() {
-            return disableInprogressNotification;
-        }
-
         public boolean isConsiderUnstableAsSuccess() {
             return considerUnstableAsSuccess;
         }
 
+        @DataBoundSetter
+        public void setConsiderUnstableAsSuccess(boolean considerUnstableAsSuccess) {
+            this.considerUnstableAsSuccess = considerUnstableAsSuccess;
+        }
+
         public String getCredentialsId() {
             return credentialsId;
         }
 
+        @DataBoundSetter
+        public void setCredentialsId(String credentialsId) {
+            this.credentialsId = StringUtils.trimToNull(credentialsId);
+        }
+
+        public boolean isDisableInprogressNotification() {
+            return disableInprogressNotification;
+        }
+
+        @DataBoundSetter
+        public void setDisableInprogressNotification(boolean disableInprogressNotification) {
+            this.disableInprogressNotification = disableInprogressNotification;
+        }
+
         public boolean isIgnoreUnverifiedSsl() {
             return ignoreUnverifiedSsl;
         }
 
+        @DataBoundSetter
+        public void setIgnoreUnverifiedSsl(boolean ignoreUnverifiedSsl) {
+            this.ignoreUnverifiedSsl = ignoreUnverifiedSsl;
+        }
+
         public boolean isIncludeBuildNumberInKey() {
             return includeBuildNumberInKey;
         }
 
+        @DataBoundSetter
+        public void setIncludeBuildNumberInKey(boolean includeBuildNumberInKey) {
+            this.includeBuildNumberInKey = includeBuildNumberInKey;
+        }
+
         public String getProjectKey() {
             return projectKey;
         }
 
+        @DataBoundSetter
+        public void setProjectKey(String projectKey) {
+            this.projectKey =  StringUtils.trimToNull(projectKey);
+        }
+
         public boolean isPrependParentProjectKey() {
             return prependParentProjectKey;
         }
 
+        @DataBoundSetter
+        public void setPrependParentProjectKey(boolean prependParentProjectKey) {
+            this.prependParentProjectKey = prependParentProjectKey;
+        }
+
+        public String getStashRootUrl() {
```

Comment:
Now that `trimToNull()` is used in `setStashRootUrl()` this can simply return the `stashRootUrl` value.